src/hip_clang.cpp

```diff
@@ -233,6 +233,9 @@ hipError_t hipLaunchByPtr(const void *hostFunction)
         " for device %d!\n", hostFunction, deviceId);
     abort();
   } else {
+    // Insert 48-bytes at the end for implicit kernel arguments and fill with value zero.
```

Comment:
kernels may be launched by hipModuleLaunchKernel. This place is not generic enough to catch all kernel launching in hip-clang. You may want to make changes to ihipModuleLaunchKernel to handle implicit kernel arg for all kernel launches.